File: back-end/src/main/kotlin/com/sorsix/skopjecarrent/model/Reservation.kt
package com.sorsix.skopjecarrent.model

import com.sorsix.skopjecarrent.model.enum.ReservationStatus
import javax.persistence.*
import java.time.LocalDateTime

@Entity
data class Reservation(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,
    @ManyToOne
    val pickUpLocation: Location,
    @ManyToOne
    val dropOffLocation: Location,
    val totalPrice: Int,
    @OneToOne
    val car: Car,
    @ManyToOne
    val client: User,
    @Enumerated(value = EnumType.STRING)
    val status: ReservationStatus,
    val pickUpDate: LocalDateTime,
    val dropOffDate: LocalDateTime,
    val reviewWritten: Boolean
)

File: front-end/src/app/model/Reservation.ts
import { ILocation } from 'src/app/model/ILocation';
import { Car } from "./Car";
import { User } from "./User";

export interface Reservation 
{
    id: number,
    pickUpLocation: ILocation,
    dropOffLocation: ILocation,
    totalPrice: number,
    car: Car,
    client: User,
    status: string,
    pickUpDate: string,
    dropOffDate: string,
    reviewWritten: Boolean
}